### ISSUE TITLE
Databases are not deleted when connections are still open

### ISSUE THREAD

**Issue by datenhahn** [ecodia/simple-database-provisioner#3]
When some application still has connections open to the db it cannot be deleted. The simple-database-provisioner fails gracefully with a descriptive error message, but it should just kill the connections to be able to clean up the db.

```
SELECT pg_terminate_backend(pg_stat_activity.pid)
FROM pg_stat_activity
WHERE pg_stat_activity.datname = 'the-db-name';
```

**Comment by Vengarioth** [comment]
this happens when someone deletes a custom resource via kubectl right? Does the controller have a chance to say no? If yes is there a way to move this behind a `--force` flag or something similar?

**Comment by datenhahn** [comment]
It happens on postgres side. SDP works, that it receives events from k8s and updates its internal state accordingly. A worker job then realizes the desired state on the DBMS. The realization fails. The k8s resource is already removed, which is the desired behaviour.

I want to keep the kubernetes handling asynchronous from the database handling. In the current setup the k8s state is fine, the only problem is that on SDP side the db will not vanish. Meaning if a similar resource is deployed again, it has to wait for the old resource to be cleaned up.

This pointless waiting is one big problem of OSBA and I need to get rid of it. The workaround is easy, as stated in the ticket: We just have to drop all connections to a db before deletion.

**Comment by datenhahn** [comment]
fixed with release v0.1.1
